src/compatTest/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPluginTests.kt

```diff
@@ -206,6 +206,102 @@ class NexusPublishPluginTests {
         assertUploadedToStagingRepo("/org/example/sample/0.0.1/sample-0.0.1.jar")
     }
 
+    @Test
+    fun `publishes to two Nexus repositories`() {
+        val otherServer = WireMockServer()
+        otherServer.start()
```

Comment:
Ah, I definitely meant to come back here and clean this up prior to opening the PR, but I totally forgot! I tried to leverage the JUnit extension for providing a new instance of a WireMockServer, but it seems to not be supported. I think we have a couple of options
1. Decorate the WireMockServer with a `Closeable` implementation that calls the appropriate shutdown methods on the server
2. Write a small junit extension to manage the lifecycle of another WireMockServer. This is nice because it lets us re-use some of the existing components of the junit5 extension already in use by this project.



This is what the `Closeable` implementation/usgae would look like


```

    class CloseableWireMockServerContainer(val port: Int, private val server: WireMockServer = WireMockServer(port)) : Closeable,
        Supplier<WireMockServer> {
        override fun close() {
            server.stop()
        }

        override fun get(): WireMockServer {
            return server
        }
    }

@Test
    fun `publishes to two Nexus repositories`() {
      CloseableWireMockServerContainer(0). use {
      
      val otherServer = it.get()
      // omitted, rest of test

  }
}
```

Or, this is what the JUnit extension would look like 
```
/**
 * Composes a [WiremockResolver] and uses that by default. But, if a parameter is annotated
 * with [MethodScopeWiremockResolver] it creates a new instance of the [WiremockResolver] extension and
 * manages its lifecycle in the scope of that [ExtensionContext]
 */
class MethodScopeWiremockResolver(
    private val inner: WiremockResolver = WiremockResolver(),
    private val localResolvers: MutableMap<String, WiremockResolver> = mutableMapOf()
) : ParameterResolver by inner, AfterEachCallback {

    /**
     * Checks to see if a method the parameter is annotated with [MethodScopedWiremockServer]. If it is, we first verify
     */
    override fun resolveParameter(parameterContext: ParameterContext, extensionContext: ExtensionContext): Any {
        return if (parameterContext.isAnnotated(MethodScopedWiremockServer::class.java)) {
            if (!parameterContext.parameter.type.isAssignableFrom(WireMockServer::class.java)) {
                throw IllegalArgumentException("Annotated type must be a WireMockServer")
            }
            return localResolvers.getOrPut(extensionContext.uniqueId, { WiremockResolver() })
                .resolveParameter(parameterContext, extensionContext)
        } else {
            inner.resolveParameter(parameterContext, extensionContext)
        }
    }

    override fun afterEach(context: ExtensionContext) {
        val localWiremockResolver = localResolvers[context.uniqueId]
        if (localWiremockResolver != null) {
            localWiremockResolver.afterEach(context)
            localResolvers.remove(context.uniqueId)
        }
        inner.afterEach(context)
    }

    /**
     * Decorates a parameter also annotated with [WiremockResolver.Wiremock]
     */
    @Target(allowedTargets = [AnnotationTarget.VALUE_PARAMETER])
    @Retention(AnnotationRetention.RUNTIME)
    annotation class MethodScopedWiremockServer
}

  @Test
    fun `publishes to two Nexus repositories`(@MethodScopedWiremockServer @Wiremock otherServer: WireMockServer) {
    // omitted, rest of test
```

Reply:
> Does JUnit Jupiter has an equivalent of `cleanup:` in Spock - to be always executed at the end of that particular test only?

It does not but it would be fairly easy to write an extension that collects `Closable` instances.

Reply:
> Does JUnit Jupiter has an equivalent of `cleanup:` in Spock - to be always executed at the end of that particular test only?

It does not but it would be fairly easy to write an extension that collects `Closable` instances.